asf_heat_pump_suitability/pipeline/prepare_features/land_extent.py

```diff
@@ -50,57 +78,52 @@ def _standardise_list_council_names(name_series: pd.Series) -> list:
     return council_names
 
 
-def generate_gdf_map_file_to_bounds(
-    land_extent_location: str = config["data_source"]["EW_inspire_land_extent"],
+def generate_gdf_file_bounds_ew(
+    path: str,
     ladnm_col: str = "LAD23NM",
     use_cols: list = ["LAD23NM", "LAD23CD", "geometry"],
-    save_as: str = None,
 ) -> gpd.GeoDataFrame:
     """
-    Generate GeoDataFrame with land extent (INSPIRE) files and their bounding polygons by matching land extent filenames
-    to ONS council polygon names. CRS: British National Grid (EPSG: 27700).
+    Generate GeoDataFrame with land extent (INSPIRE) filenames for England and Wales and their bounding polygons.
+    CRS: British National Grid (EPSG: 27700).
 
     Args:
-        land_extent_location (str): location of land extent (INSPIRE) files. Defaults to S3 location.
+        path (str): S3 location of land extent (INSPIRE) files.
         ladnm_col (str): name of column with council (LAD) names in council polygons file
         use_cols (list): names of columns to keep in council polygons file. Must include the following columns:
         council/LAD name, council/LAD code, council/LAD geometry.
-        save_as (str): path to save matched files to. Optional.
 
     Returns:
         gpd.GeoDataFrame: land extent (INSPIRE) files and their bounding polygons in British National Grid CRS
     """
-    council_bounds = transform_gdf_council_bounds(ladnm_col, use_cols)
-    land_extent_file_names = base_getters.list_files_s3_location(land_extent_location)
+    bounds_gdf = transform_gdf_council_bounds(ladnm_col, use_cols)
+    files = base_getters.list_obj_s3_location(path)
 
     matches = _match_list_file_to_name(
```

Comment:
Thank you , this is a great suggestion which I have added. I kept the `fill_null_file_bounds` function in the function to generate file bounds because I think the INSPIRE files are missing a few local authorities at the source, so those can be filled if needed if they ever get added and we re-run this pipeline.

I also added a new assertion statement to ensure that the pipeline errors out if there are more than 10% of .gml files with no file bounds.